### PR TITLE
Feature slack notification

### PR DESCRIPTION
**Pull request by julian9499**
This resolves issue #480 and enables #237 to be solved through slack.

### CODE REVIEW

File: src/main/java/ch/wisv/areafiftylan/exception/SlackNotificationException.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+package ch.wisv.areafiftylan.exception;
+
+
+import java.io.IOException;
+
+public class SlackNotificationException extends AreaFiftyLANException {
+    public SlackNotificationException(IOException e) {
+        super("Unable to send a notification to slack error:" + e.getMessage());
```
Comment (martijnjanssen) [inline src/main/java/ch/wisv/areafiftylan/exception/SlackNotificationException.java]:
```suggestion
        super("Unable to send a notification to slack:" + e.getMessage());
```

File: src/main/java/ch/wisv/areafiftylan/products/service/MolliePaymentService.java [hunk 2]
```diff
@@ -131,6 +134,7 @@ public Order updateStatus(String orderReference) {
                     }
                     case "paid": {
                         order.setStatus(OrderStatus.PAID);
+                        slackNotificationService.sendSlackMessage(order.getTickets().size() + " tickets bought!");
```
Comment (martijnjanssen) [inline src/main/java/ch/wisv/areafiftylan/products/service/MolliePaymentService.java]: We may need a :tada: or :money_with_wings: emoji in here

Comment (julian9499) [reply src/main/java/ch/wisv/areafiftylan/products/service/MolliePaymentService.java]: Implemented in the Aspect!

File: src/main/java/ch/wisv/areafiftylan/utils/SlackNotificationAspect.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    @After("execution(* ch.wisv.areafiftylan.products.service.TicketServiceImpl.transferTicket(..))")
+    public void notifyTicketTransfer() {
+        sendSlackMessage("A ticket transfer has been performed");
+    }
+
+    @AfterThrowing(pointcut = "execution(* ch.wisv.areafiftylan.exception.AreaFiftyLANException.AreaFiftyLANException(..))", throwing = "ex")
+    public void forwardException(Throwable ex) {
+        sendSlackMessage("An exception has been thrown: " + ex.getMessage());
```
Comment (martijnjanssen) [inline src/main/java/ch/wisv/areafiftylan/utils/SlackNotificationAspect.java]: Wait, so we're sending slack notification errors to slack? That seems like a bad idea to be honest.

Comment (julian9499) [reply src/main/java/ch/wisv/areafiftylan/utils/SlackNotificationAspect.java]: I added an exception to the slackException

File: src/main/java/ch/wisv/areafiftylan/exception/AreaFiftyLANException.java [hunk 4]
```diff
@@ -30,6 +30,8 @@ public AreaFiftyLANException(LogLevelEnum logEnum, String message) {
         logEnum.logMessage(message);
     }
 
+
```
Comment (martijnjanssen) [inline src/main/java/ch/wisv/areafiftylan/exception/AreaFiftyLANException.java]: Let's remove these.

File: src/main/java/ch/wisv/areafiftylan/utils/SlackNotificationAspect.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    public void notifyTicketTransfer() {
+        sendSlackMessage("A ticket transfer has been performed");
+    }
+
+    @AfterThrowing(pointcut = "execution(* ch.wisv.areafiftylan.exception.*.*(..)) && !execution(* ch.wisv.areafiftylan.exception.SlackNotificationException.*(..))", throwing = "ex")
+    public void forwardException(Throwable ex) {
+        sendSlackMessage("An exception has been thrown: " + ex.getMessage());
+    }
+
+
```
Comment (martijnjanssen) [inline src/main/java/ch/wisv/areafiftylan/utils/SlackNotificationAspect.java]: And these 2 newlines :)

File: src/main/java/ch/wisv/areafiftylan/exception/AreaFiftyLANException.java (hunk 4 above)
Comment (julian9499) [reply src/main/java/ch/wisv/areafiftylan/exception/AreaFiftyLANException.java]: done

File: src/main/java/ch/wisv/areafiftylan/utils/SlackNotificationAspect.java (hunk 5 above)
Comment (julian9499) [reply src/main/java/ch/wisv/areafiftylan/utils/SlackNotificationAspect.java]: done

**Pull request merged**